android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -69,7 +74,14 @@ constructor(
     val configKey = if (configType.multiConfig && configId != null) configId else configType.name
     return if (configType.parseAsResource)
       configsJsonMap.getValue(configKey).decodeResourceFromString()
-    else configsJsonMap.getValue(configKey).decodeJson(jsonInstance = json)
+    else
+      LocaleUtil(this)
```

Comment:
I understand why we initialize `LocaleUtil` via it's constructor here because we would have cyclic dependency during injection since, the class needs `ConfigurationRegistry`. Let's make `LocaleUtil` regular class by removing the dependency injection annotations. We can initialize it as a `lazy { }` delated property in `ConfigurationRegistry` to make it singleton within the registry class.